### PR TITLE
Remove reliance on EPPO download by using GBIF API

### LINKED ISSUE #590: EPPO download not working
For the last week or two the EPPO data download from https://data.eppo.int/files/xmlfull.zip has failed mid-stream. We have had other issues with using EPPO, so we might consider replacing it with GBIF data.

The only places we use their data are in `weedcoco.utils.get_supercategory_names` and `weedcoco.validation.check_plant_species_format` to get ancestry of a taxon, and to check validity of a taxon.

We can instead adopt GBIF... Port the code from [the frontend](https://github.com/Weed-AI/Weed-AI/blob/1d5ec5c1f0a73a49003ebbac5f673a8f39eed3f9/search/src/Components/search/CategoryTooltip.js#L40-L93) to Python looks something like:
```py
def get_ancestry(name):
    results = requests.get(f"https://api.gbif.org/v1/species?name={urlencode(name)}&datasetKey=d7dddbf4-2cf0-4f39-9b2a-bb099caae36c&taxonomicStatus=ACCEPTED")
    assert results["endOfRecords"]  # TODO?: pagination
    gbif_record = next(record for record in results["results"]
                       if record["canonicalName"].lower() == name
                       and record["taxonomicStatus"] == "ACCEPTED")

    return {rank: record[rank]
            for rank in ["kingdom", "phylum", "order", "family", "genus", "species"]
            if rank in record}
```

### PR DESCRIPTION
**Pull request by jnothman**
This replaces our uses of EPPO with (cached) calls to the GBIF API

Fixes #590
Resolves #559

Disadvantages of the new approach include:
* requires an internet connection
* could be exploited to spam GBIF and we don't currently have any protection against this (e.g. limiting the number of categories in a dataset)
* does not support variants as species names, i.e. while EPPO recognised "brassica oleracea var. alboglabra" (gai lan) as a subspecies of "brassica oleracea" (broccoli, cabbage, etc.), GBIF does not have a taxon for the variant. This is more likely a concern for crop species than weed species.
* may make it harder to get mappings to EPPO codes in the future.

It makes our server and continuous integration more reliable, however.

Are you concerned by any of the above, @geezacoleman?

### CODE REVIEW

**Comment by geezacoleman** [comment]
If EPPO just keeps failing then it's much better to have taxon support at all. The impact on variants is most likely the biggest loss, particularly for horticultural systems because they can look so different. Not so concerned about it for large-scale crops.

**Comment by jnothman** [comment]
This seems to fix the issue we had on other PRs where Cypress broke while saving a category mapping and hence validating species names. A new issue appears in Cypress, however, where it appears setting the metadata form from pasted JSON is not working... Quite surprising. Any ideas, @ElevnLi?

**Comment by jnothman** [comment]
When I run Cypress locally, I'm getting a different error, currently...

**Comment by jnothman** [comment]
And when I run it with `yarn cypress open` I'm not getting any error locally. I wonder what I need to do to reproduce it...

**Comment by jnothman** [comment]
Both the CI and local tests are running with 
```
┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    7.2.0                                                                              │
  │ Browser:    Electron 89 (headless)                                                             │
  │ Specs:      1 found (weedai_upload.js)                                                         │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
```
So I'm not sure where to go next to debug.

**Comment by jnothman** [comment]
It puzzles me that not only does the test failing in CI succeed on my local machine, but that the exact same code is used for the VOC test and it works (in CI and locally).

**Comment by jnothman** [comment]
I managed to reproduce this on my laptop in a clean setup, trying to emulate what happens on CI. I've not managed to work out what the difference is between where it was working and where it was failing yet.

**Comment by jnothman** [comment]
The issue here appears intermittently. It will be interesting to see if it is intermittent on master.

**Comment by jnothman** [comment]
Did you land up using this when Eppo was causing you pain, @ElevnLi? Should we just merge it and forget about EPPO?

**Review by hlydecker — COMMENTED** [review]
This greatly simplifies things; good stuff

File: search/django/weedid/views.py [hunk 1]
```diff
@@ -529,11 +543,4 @@ def sitemap_xml(request):
...
         request, "sitemap.xml", context={"urls": urls}, content_type="text/xml"
-    )
-
-
-def warmup(request):
-    from weedcoco.validation import EPPO_CACHE_PATH, get_eppo_singleton
-
-    get_eppo_singleton(EPPO_CACHE_PATH)
-    return HttpResponse("Success")
+    )
```
Comment (hlydecker) [inline search/django/weedid/views.py:546]:
```suggestion
    )
    
```

**Pull request merged**